src/modules/stream/internal.js

```diff
@@ -97,8 +129,12 @@ export async function internalStream(streamInfo, res) {
         if (req.statusCode < 200 || req.statusCode > 299)
             return res.end();
 
-        req.body.pipe(res);
-        req.body.on('error', () => res.end());
+        if (["application/vnd.apple.mpegurl", "audio/mpegurl"].includes(req.headers['content-type'])) {
```

Comment:
https://vimeo.com/288386543 with 2160p quality returns a blank file.

adding correct content type to list of types results in `Assertion failed` console log and infinitely hanging stream:
<img width="164" alt="image" src="https://github.com/imputnet/cobalt/assets/71202418/393bafed-e7da-44db-bb25-10bee91deff9">


before PR it works fine, please test all cases.

Reply:
That meme... I'm dead.